src/ape/types/__init__.py

```diff
@@ -40,15 +45,136 @@
 """
 
 
-@dataclass
-class ContractLog:
+class TopicFilter(BaseModel):
+    event: EventABI
+    search_values: Dict[str, Optional[Union[Any, List[Any]]]] = {}
+
+    @property
+    def event_signature_hash(self) -> str:
+        return add_0x_prefix(HexStr(keccak(text=self.event.selector).hex()))
+
+    @root_validator(pre=True)
+    def validate_search_values(cls, values):
+        from ape.utils import is_dynamic_sized_type
+
+        values["event"] = (
+            values["event"].abi if hasattr(values["event"], "abi") else values["event"]
+        )
+        input_types = {i.name: i.type for i in values["event"].inputs}
+
+        def encode_topic_value(key, value):
+            if hasattr(value, "address"):
+                value = value.address
+
+            abi_type = input_types.get(key)
+
+            if not abi_type or value is None:
+                return None
+
+            elif isinstance(value, (list, tuple)):
+                return [encode_topic_value(key, v) for v in value]
+
+            elif is_dynamic_sized_type(abi_type):
+                return to_hex(keccak(encode_single_packed(str(abi_type), value)))
```

Comment:
`to_hex` main purpose seems encoding numeric values for jsonrpc. when dealing with bytes, it just delegates to `encode_hex`, so we could just it instead
```suggestion
                return encode_hex(keccak(encode_single_packed(str(abi_type), value)))
```